oomph_pygments_lexer.py

```diff
@@ -0,0 +1,268 @@
+            # backslashes, quotes and formatting signs must be parsed one at a time
+            (r'[^\\\'"{}\n]+', ttype),
+            (r'[\'"\\]', ttype),
+            # newlines are an error (use "nl" state)
+        ]
+
+    tokens = {
+        "root": [
+            (r"\n", Text),
+            (
+                r'^(\s*)([rRuUbB]{,2})("""(?:.|\n)*?""")',
+                bygroups(Text, String.Affix, String.Doc),
+            ),
+            (
+                r"^(\s*)([rRuUbB]{,2})('''(?:.|\n)*?''')",
+                bygroups(Text, String.Affix, String.Doc),
+            ),
+            (r"\A#!.+$", Comment.Hashbang),
+            (r"#.*$", Comment.Single),
+            (r"\\\n", Text),
+            (r"\\", Text),
+            include("keywords"),
+            (r"(def)((?:\s|\\\s)+)", bygroups(Keyword, Text), "funcname"),
+            (r"(class)((?:\s|\\\s)+)", bygroups(Keyword, Text), "classname"),
+            (r"(from)((?:\s|\\\s)+)", bygroups(Keyword.Namespace, Text), "fromimport"),
+            (r"(import)((?:\s|\\\s)+)", bygroups(Keyword.Namespace, Text), "import"),
+            include("expr"),
+        ],
+        "expr": [
+            # non-raw f-strings
+            (
+                '(""")',
+                bygroups(String.Double),
+                combined("fstringescape", "tdqf"),
+            ),
+            (
+                "(''')",
+                bygroups(String.Single),
+                combined("fstringescape", "tsqf"),
+            ),
+            (
+                '(")',
+                bygroups(String.Double),
+                combined("fstringescape", "dqf"),
+            ),
+            (
+                "(')",
+                bygroups(String.Single),
+                combined("fstringescape", "sqf"),
+            ),
+            (r"[^\S\n]+", Text),
+            (r"!=|==|<<|>>|:=|[-~+/*%=<>&^|.]", Operator),
+            (r"[]{}:(),;[]", Punctuation),
+            (r"(in|is|and|or|not)\b", Operator.Word),
+            include("builtins"),
+            include("name"),
+            include("numbers"),
+        ],
+        "expr-inside-fstring": [
+            (r"[{([]", Punctuation, "expr-inside-fstring-inner"),
+            # without format specifier
+            (
+                r"(=\s*)?"  # debug (https://bugs.python.org/issue36817)
+                r"(\![sraf])?"  # conversion
+                r"\}",
+                String.Interpol,
+                "#pop",
+            ),
+            # with format specifier
+            # we'll catch the remaining '}' in the outer scope
+            (
+                r"(=\s*)?"  # debug (https://bugs.python.org/issue36817)
+                r"(\![sraf])?"  # conversion
+                r":",
+                String.Interpol,
+                "#pop",
+            ),
+            (r"\s+", Text),  # allow new lines
+            include("expr"),
+        ],
+        "expr-inside-fstring-inner": [
+            (r"[{([]", Punctuation, "expr-inside-fstring-inner"),
```

Comment:
Braces inside f-strings are `\{` oomph and `{{` in python. Don't know where to put this though.